src/plugins/integrations/gradle.ts

```diff
@@ -0,0 +1,38 @@
+import { readFileSync } from 'fs';
+import { SiloDiscoveryConfig } from '../types';
+
+const regex = [/:(.\S*):/, /name: ?'(.*?)'/];
+
+export const gradle: SiloDiscoveryConfig = {
+  supportedFiles: ['build.gradle', 'build.gradle.kts'],
+  ignoreDirs: [
+    'gradle-app.setting',
+    'gradle-wrapper.jar',
+    'gradle-wrapper.properties',
+  ],
+  scanFunction: (filePath) => {
+    const lines = readFileSync(filePath)
+      .toString()
+      // split on new line character
+      .split(String.fromCharCode(10));
+
+    const deps: string[] = [];
+    lines.map((line) => {
+      let rExp = null;
+      // eslint-disable-next-line no-restricted-syntax
+      for (const reg of regex) {
```

Comment:
This is a bit hard to read. Maybe you can rewrite using the `regexes.find((reg) => new RegExp(reg, 'g').test(line))` syntax?